src/prometheus_alert_rules/general.yaml

```
@@ -0,0 +1,13 @@
+groups:
+- name: HardwareObserver
+  rules:
+    - alert: CollectorFailed
+      expr: '{__name__=~"(.*)_collector_failed"} == 1'
+      for: 0m
+      labels:
+        severity: warning
```

If collector failed it should not be critical instead of just warning?

---

Critical for me is more like hardware issue.
It's more like a internal bug we can fix.

---

Updated to critical

---

I think it's all about how alarmed we want the cloud operator to be in case of this issue. IMO if the collector itself failed (which I hope wouldn't happen often), the error should be `critical` since there's no metrics now being exported from that collector and the operator should be notified so they can raise some issue against it.

---

My judgeent is 

"Critical":  It's an issue that 90+% will impact the service
"Warning": It's an issue but not so critical. It probably won't impact the service immediately. 

For this case, I think COE just need to restart the service. It won't impact the service immediately in general case. Therefore, I vote warning. 

Furthermore, they can use customize rule to overwrite the severity when it is `warning`.  (But they can't if we set it as critical). Therefore, before we have confirmed answer, it is probably good enough to set it as `warning` . 

Need opinion from @aieri or @przemeklal

---

If collector failed it can be issue on hw-observer, but it can mean that something is not working on hardware -> getting metrics failed. To me this is critical or better would be "unknown" like we had in nagios, but that's not possible.
Regarding alert rule I think alert should be fired only if there are multiple failure in X minutes, since now there is no need to restart hw-observer.

---

In #100 , we would like to follow the dynamic notifications. I will suggest to set severity to `error` from it's defination.